Src/AutoFakeItEasy/MethodRule.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using AutoFixture.Kernel;
+using FakeItEasy.Core;
+
+namespace AutoFixture.AutoFakeItEasy
+{
+    /// <summary>
+    /// A rule that intercepts method calls. Supplies the return and all out and ref values
+    /// from the fixture. New values will be fetched from the fixture on each call.
+    /// </summary>
+    internal class MethodRule : IFakeObjectCallRule
+    {
+        private readonly ISpecimenContext context;
+
+        public MethodRule(ISpecimenContext context)
+        {
+            this.context = context;
+        }
+
+        public int? NumberOfTimesToCall { get; } = null;
```

Comment:
Why do we have it? Did it come from `IFakeObjectCallRule` ? If so, then please:
- make it lambda;
- add a comment 😉 

Same for `PropertyRule`.